### PR TITLE
New global settings client and a new fixture method to filter by gameweek

### PR DESCRIPTION
**Pull request by RyanDurkin**
I'm having a play around with querying the FPL API so I can email myself when one of my players scores/assists/booked etc and added these extra pieces of functionality for it.

### CODE REVIEW

File: src/FplClient/IFplGlobalSettingsClient.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading.Tasks;
+using FplClient.Data;
+
+namespace FplClient
+{
+    public interface IFplGlobalSettings
```
Comment (RagtimeWilly) [inline src/FplClient/IFplGlobalSettingsClient.cs]: Interface doesn't match file name here

File: src/FplClient/IFplFixtureClient.cs [hunk 2]
```diff
@@ -7,5 +7,7 @@ namespace FplClient
     public interface IFplFixtureClient
     {
         Task<IEnumerable<FplFixture>> GetFixtures();
+
+        Task<IEnumerable<FplFixture>> GetFixtures(int eventId);
```
Comment (RagtimeWilly) [inline src/FplClient/IFplFixtureClient.cs]: Would `GetFixturesByGameweek` be a better method name here?

File: src/FplClient/IFplGlobalSettingsClient.cs (hunk 1 above)
Comment (RagtimeWilly) [inline src/FplClient/IFplGlobalSettingsClient.cs]: This should be called `IFplGlobalSettingsClient`

File: src/FplClient.Tests/Clients/FplFixtureClientTests.cs [hunk 3]
```diff
@@ -24,6 +24,14 @@ public void Retrieves_fixtures_successfully()
             Assert.AreEqual(380, data.Count());
         }
 
+        [Test]
+        public void Retrieves_fixtures_gameweek_one_successfully()
```
Comment (RagtimeWilly) [inline src/FplClient.Tests/Clients/FplFixtureClientTests.cs]: `Retrieves_fixtures_for_gameweek_one_successfully`

File: src/FplClient/Clients/FplFixtureClient.cs [hunk 4]
```diff
@@ -30,5 +30,19 @@ public FplFixtureClient(Func<HttpClient> clientFactory)
...
             }
         }
+
+        public async Task<IEnumerable<FplFixture>> GetFixtures(int eventId)
+        {
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+
+            using (var client = _clientFactory())
+            {
+                string url = "https://fantasy.premierleague.com/drf/fixtures/?event=" + eventId;
```
Comment (RagtimeWilly) [inline src/FplClient/Clients/FplFixtureClient.cs]: Could use string interpolation here?

**Review by RagtimeWilly — COMMENTED** [review]
Looks great! Some minor comments around naming.

**Comment by RyanDurkin** [comment]
Thanks for the review. Code has been updated.

**Pull request merged**